CMakeLists.txt

```diff
@@ -1,38 +1,27 @@
 project(charon NONE)
-cmake_minimum_required(VERSION 3.6) #Tested only with 3.6.1 and 3.9.1.
+cmake_minimum_required(VERSION 3.18)
 
-# FIXME: Remove the code for CMake <3.12 once we have switched over completely.
-# FindPython3 is a new module since CMake 3.12. It deprecates FindPythonInterp and FindPythonLibs.
-if(${CMAKE_VERSION} VERSION_LESS 3.12)
-    # Use FindPythonInterp and FindPythonLibs for CMake <3.12
-    find_package(PythonInterp 3.4 REQUIRED)
-
-    set(Python3_EXECUTABLE ${PYTHON_EXECUTABLE})
-    set(Python3_VERSION_MAJOR ${PYTHON_VERSION_MAJOR})
-    set(Python3_VERSION_MINOR ${PYTHON_VERSION_MINOR})
-else()
-    # Use FindPython3 for CMake >=3.12
-    find_package(Python3 ${CURA_PYTHON_VERSION} EXACT REQUIRED COMPONENTS Interpreter)
+if(NOT DEFINED Python_VERSION)
+    set(Python_VERSION
+            3.10
+            CACHE STRING "Python Version" FORCE)
+    message(STATUS "Setting Python version to ${Python_VERSION}. Set Python_VERSION if you want to compile against an other version.")
+endif()
+if(APPLE)
+    set(Python_FIND_FRAMEWORK NEVER)
+endif()
+find_package(Python ${Python_VERSION} EXACT REQUIRED COMPONENTS Interpreter)
+message(STATUS "Linking and building ${project_name} against Python ${Python_VERSION}")
+if(NOT DEFINED Python_SITELIB_LOCAL)
```

Comment:
Where/how's this expected to be set now?

Reply:
Cura-build sets this variable upon building. The reason is that all other packages are installed to `site-packages` on all platforms (Windows, MacOS and Linux) but on Debian-based platforms like our Ubuntu build server the default for `Python_SITELIB` is `dist-packages`.

We added this so that from cura-build we could override the installation directory to be consistent so that we would have to make fewer exceptions.

Reply:
Since these changes are aimed at changes/requirements from the cura-build, is it stil possible to use this for a cura-build-less build of libCharon?

Reply:
No, that's not our intention. That would be very unworkable from you!

If `Python_SITELIB_LOCAL` is not specified it should default to the old behaviour.

Would it be acceptable if we add documentation to this about that this parameter exists and leave the behaviour as-is, then?